src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -137,6 +149,64 @@ void AndroidDeviceControllerWrapper::OnPairingDeleted(CHIP_ERROR error)
     CallJavaMethod("onPairingDeleted", static_cast<jint>(error));
 }
 
+CHIP_ERROR AndroidDeviceControllerWrapper::GenerateNodeOperationalCertificate(const chip::PeerId & peerId, const chip::ByteSpan & csr,
+                                                                                int64_t serialNumber, uint8_t * certBuf,
+                                                                                uint32_t certBufSize, uint32_t & outCertLen)
```

Comment:
Suggest adding a TODO to refactor this API to match latest spec, so that GenerateNodeOperationalCertificate receives the full CSR Elements data payload, so that signature can be checked. Can be done in a future refactor.

Reply:
Added.